internal/installercache/installercache.go

```diff
@@ -138,14 +140,25 @@ func (i *Installers) Get(ctx context.Context, releaseID, releaseIDMirror, pullSe
 	for {
 		select {
 		case <-ctx.Done():
-			return nil, ctx.Err()
+			err := ctx.Err()
+			if err == context.DeadlineExceeded {
+				i.metricsAPI.InstallerCacheGetReleaseTimeout(releaseID)
+			}
+			return nil, err
 		default:
-			release, err := i.get(releaseID, releaseIDMirror, pullSecret, ocRelease, ocpVersion, clusterID)
+			majorMinorVersion, err := ocRelease.GetMajorMinorVersion(i.log, releaseID, releaseIDMirror, pullSecret)
+			if err != nil {
+				i.log.Warnf("unable to get majorMinorVersion to record metric for %s falling back to full URI", releaseID)
+				majorMinorVersion = releaseID
+			}
+			release, err := i.get(releaseID, releaseIDMirror, pullSecret, ocRelease, ocpVersion, clusterID, majorMinorVersion)
 			if err == nil {
+				i.metricsAPI.InstallerCacheGetReleaseOK(majorMinorVersion)
```

Comment:
We agreed on having 2 metrics:
* `assisted_installer_cache_get_release{hit="(true|false)"}`
*  `assisted_installer_cache_get_release_error{error="(timeout|whatever|other/unknown/undefined)"}`

I still think the error metric won't be of much (or any) use, but definitely we do not want to count all requests there. What I thought of this is hits+misses+errors should return the total number of requests to this function. 

Did I misunderstand anything?

Reply:
I think you did, I thought we were at a consensus that we would record both